Source/WebCore/html/HTMLElement.cpp

```diff
@@ -1456,7 +1456,8 @@ ExceptionOr<void> HTMLElement::hidePopoverInternal(FocusPreviousElement focusPre
 
     ASSERT(popoverData());
 
-    removeFromTopLayer();
+    if (isInTopLayer())
+        removeFromTopLayer();
```

Comment:
Is there a particular reason we need this change? (and would the spec need changing as well?)

Reply:
I saw the ASSERT in removeFromTopLayer happen when hiding the popover on cleanup/node removal. I think this can happen when dialog and popover API are intermixed. Probably the popover+dialog is shown, the popover+dialog closed and then removed. Note thay HTMLDialogElement::close tests for isInTopLayer.

Reply:
I think I see the issue here, `dialog.close()` is also removing the popover from the top layer. Not sure this is intended. Perhaps we should only remove the popover from the top layer if the "is modal" flag is set to true in `HTMLDialogElement::close()`. Will suggest a spec change.

Reply:
@rwlbuis Can you check if the latest spec change: https://github.com/whatwg/html/pull/9344 resolves the issue without the need for changing `hidePopover()` ?

Reply:
It seems to work in my release build, but I did not try debug yet. Anyway hopefully mac-AS-debug-wk2 will show whether the new patch asserts/works or not.